src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/JsonDictionaryConverter.cs

```diff
@@ -159,6 +160,7 @@ protected static JsonConverter<T> GetConverter<T>(JsonTypeInfo typeInfo)
                     }
 
                     state.Current.ObjectState = StackFrameObjectState.StartToken;
+                    state.Current.JsonPropertyInfo = elementTypeInfo.PropertyInfoForTypeInfo;
```

Comment:
Is the value different or is this a no-op?

Reply:
It is -- up to that point the value of `JsonPropertyInfo` is just the default one for the parent type:

https://github.com/dotnet/runtime/blob/11b79618faf1023ba4ea26b4037f495f81070d79/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReadStack.cs#L155